code/controllers/subsystems/chat.dm

```diff
@@ -0,0 +1,87 @@
+SUBSYSTEM_DEF(chat)
+	name = "Chat"
+	flags = SS_TICKER
+	wait = 1 // SS_TICKER means this runs every tick
+	priority = FIRE_PRIORITY_CHAT
+	init_order = INIT_ORDER_CHAT
+
+	var/list/msg_queue = list()
+
+/datum/controller/subsystem/chat/Initialize(timeofday)
+	init_vchat()
+	..()
+
+/datum/controller/subsystem/chat/fire()
+	var/list/msg_queue = src.msg_queue // Local variable for sanic speed.
+	for(var/i in msg_queue)
+		var/client/C = i
+		var/list/messages = msg_queue[C]
+		msg_queue -= C
+		if (C)
+			C << output(jsEncode(messages), "htmloutput:putmessage")
```

Comment:
Not sure how chat messages are 'handled' with the stream operator (aka `client << "message"`), since it's an internal byond function. If the MC is totally destroyed but the server is not frozen, then chat messages would stop with the advent of SSchat. In SSchat-world or pre-SSchat, messages would stop if the server was deadlocked either way.

Movement would continue (until someone ports SSinput anyway). But the game would be unplayable: machinery would no longer function, mobs would not get life() ticks, etc. I would not worry too much about chat messages in a world where the MC is not running, because that would be a short-lived round anyway.

This did happen once downstream, before we realized the bug with MC and services that have both `SS_BACKGROUND` and `SS_NO_TICK_CHECK` crashing the MC (which is the case on the inactivity subsystem until we changed it). People wrote things on paper and passed them back and forth. It was funny, though the round couldn't really continue because the MC was gone and things like doors and stuff no longer functioned since they are machinery. Restarting the MC fixes the issue and sends all enqueued messages.